xla/service/scatter_determinism_expander.cc

```diff
@@ -0,0 +1,459 @@
+/* Copyright 2024 The OpenXLA Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "xla/service/scatter_determinism_expander.h"
+#include <cstdint>
+#include "absl/container/flat_hash_set.h"
```

Comment:
You will also need a corresponding BUILD dependency "@com_google_absl//absl/container:flat_hash_set"

Reply:
changed